### DISCUSSION TITLE
How to add GTM on OpnForm to follow the form sent ?

### DISCUSSION THREAD

**Discussion by Claude-27** [OpnForm/OpnForm#111] [General]
Hello,
I have a client who made a website and use you form, (who are really usefull 👍). I would like to create a GAds campaign but I need to know eveytime or someone send a form complete. For that I use Google Tag Manager. 
How can I add GTM to my OpnForm ?
Thanks for your answer

**Comment by Claude-27** [comment]
I found, I just to intégrate my GTM code in the Custom Code ;)

**Reply by JhumanJ** [reply]
Hey, exactly that's the way to go. You can also track events if you need to by adding custom code like this:
```
<script>
window.addEventListener('message', function(event) {
   // Make sure that the event data has the type `form-submitted`
   if (event.data?.type !== 'form-submitted' ||  event.data?.form?.slug != <FORM_SLUG>) {
       return
   }    
   // Do something (redirect, track your own event, etc)
   // ...
});
</script>
```
